File: scripts/services/google_cloud.py
from core.utils import make_event, make_service

SERVICE = {
    "id": "google-cloud",
    "name": "Google Cloud",
    "desc": "Google 雲端運算與數據分析平台",
    "category": "cloud",
    "page_url": "https://status.cloud.google.com",
    "data_url": "https://status.cloud.google.com/incidents.json",
}

def fetch(ctx):
    try:
        data = ctx["http"].get_json(SERVICE["data_url"])
        if not isinstance(data, list):
            return make_service(SERVICE, "parse_failed", "official", [])

        active = [x for x in data if not x.get("end")]
        source = (active if active else data)[: ctx["config"]["max_events"]]
        events = []

        for inc in source:
            title = inc.get("external_desc") or inc.get("service_name") or ""
            events.append(
                make_event(
                    title,
                    "resolved" if inc.get("end") else "investigating",
                    inc.get("begin"),
                    inc.get("end"),
                    SERVICE["page_url"],
                )
            )

        return make_service(SERVICE, "ok" if events else "no_events", "official", events)
    except Exception as e:
        if ctx["config"].get("debug"):
            print("GCP", e)
        return make_service(SERVICE, "fetch_failed", "none", [])

File: assets/services/google-cloud.js
/* CloudStatus service module: Google Cloud */
(function () {
  "use strict";
  window.CloudStatusServices.register({
  "id": "google-cloud",
  "name": "Google Cloud",
  "desc": "Google 雲端運算與數據分析平台",
  "category": "cloud",
  "page": "https://status.cloud.google.com",
  "parser": "google-cloud",
  "sources": [
    {
      "type": "gcp",
      "url": "https://status.cloud.google.com/incidents.json",
      "label": "官方 JSON",
      "tier": 20,
      "kind": "official-json",
      "priority": 20
    },
    {
      "type": "reader",
      "label": "官方狀態頁",
      "tier": 50,
      "kind": "official-status",
      "priority": 50
    }
  ]
});
})();
